mailserver.env

```diff
@@ -260,7 +260,10 @@ ENABLE_SPAMASSASSIN=0
 # deliver spam messages in the inbox (eventually tagged using SA_SPAM_SUBJECT)
 SPAMASSASSIN_SPAM_TO_INBOX=1
 
-# spam messages will be moved in the Junk folder (SPAMASSASSIN_SPAM_TO_INBOX=1 required)
+# only has an effect if `ENABLE_SPAMASSASSIN=1`
```

Comment:
I was thinking about how to do this as well. I think ideally, only the real documentation should contain the descriptions, etc. and `mailserver.env` should have only the variables and the default values.

But I agree, this should not be a blocker for this PR. Anything else that comes to mind when reviewing this PR? :)

Reply:
I haven't time yet to review the whole PR properly. That's just one thing that I immediately saw. I will do it later or tomorrow.

> I think ideally, only the real documentation should contain the descriptions

I slightly disagree, we should stick to one decision. My "thats not something this PR must solve" relates only the "PS:" part.

General options for the future could be:

1. Include the description (pro: you don't have to open the documentary).
2. Or purge all description, and link to the documentation.

But we should try to avoid a mixed setup. In the last PRs I always put the same description in the env file as in the documentation. That said, I'd like to see here the description as well, until we decide to go another route.

Reply:
I will add a description here as well. We can open a maintainers discussion about this if want to have a decision on the topic.

Reply:
I haven't looked over the PR to review it, but glanced over this conversation.

Presently the ENV file and docs aren't in sync, some have information or vars that the other lacks.

There is a syntax feature for the docs which would be suitable for documenting these vars and should generate semantic HTML for a parser tool to inspect and extract out the env details fairly easily from that point that we could generate our `mailserver.env` from it if desired.

I'd offer to tackle that but already have a lot on my backlog for this project to handle that's dependent upon me restoring my PC (_waiting on a new monitor to be delivered from overseas, as current one seemed to have gotten damaged from powercut or being a decade old (still used DVI instead of HDMI / DP)_). Happy to take care of the env generation once I clear out my existing backlog for the project though :sweat_smile:

Reply:
That sounds great @polarathene! There is, as always, no timing constraint of course - you know that, just wanted to say that again :) I will resolve this conversation as we have reached a plan for the future.